lib/phase/flatten.js

```diff
@@ -19,31 +43,90 @@ function flattenSharedIntoJobs(shared, jobs) {
         const oldJob = clone(jobs[jobName]);
 
         // Replace
-        ['image', 'matrix', 'steps'].forEach((key) => {
+        ['image', 'matrix', 'steps', 'template'].forEach((key) => {
             if (oldJob[key]) {
                 newJob[key] = oldJob[key];
             }
         });
 
-        if (!newJob.environment) {
-            newJob.environment = {};
-        }
-        if (!newJob.secrets) {
-            newJob.secrets = [];
-        }
-        if (!newJob.settings) {
-            newJob.settings = {};
+        merge(newJob, oldJob);
+        newJobs[jobName] = newJob;
+    });
+
+    return newJobs;
+}
+
+/**
+ * Retrieve template and merge into job config
+ *
+ * @method mergeTemplateIntoJob
+ * @param  {String}     jobName         Job name
+ * @param  {Object}     jobConfig       Job config
+ * @param  {Object}     newJobs         List of new jobs
+ * @param  {String}     apiUrl          API Url to retrieve templates
+ * @return {Promise}
+ */
+function mergeTemplateIntoJob(jobName, jobConfig, newJobs, apiUrl) {
+    const matched = TEMPLATE_REGEX.exec(jobConfig.template);
+    const name = matched[NAME_MATCH];
+    const version = matched[VERSION_MATCH];
+    const breaker = new Fusebox(request, {});
+    const oldJob = jobConfig;
+
+    // Try to get the template
+    return breaker.runCommand({
+        url: `${apiUrl}/v4/templates?name=${name}&version=${version}`,
```

Comment:
Why are we using query parameters here?  They cannot be cached.  Shouldn't this be `/templates/$name/$version/`?

Also, shouldn't we be escaping them?